### PR TITLE
Document use of `allow_concurrent_runs` with long-running tasks

### PR DESCRIPTION
**Pull request by markuswustenberg**
See https://github.com/zed-industries/zed/issues/17950

Release Notes:

- N/A

### CODE REVIEW

**Comment by markuswustenberg** [comment]
@cla-bot check

**Review by SomeoneToIgnore — COMMENTED** [review]
I propose to create a separate header below the JSON, or even just a sentence below, saying explicitly which two parameter values we have to set.

The new change looks somewhat more confusing than before to me.

**Comment by markuswustenberg** [comment]
Gotcha, will have a look tomorrow.

**Comment by markuswustenberg** [comment]
@SomeoneToIgnore better?

File: docs/src/tasks.md [hunk 1]
```diff
@@ -50,6 +50,8 @@ There are two actions that drive the workflow of using tasks: `task: spawn` and
 `task: spawn` opens a modal with all available tasks in the current file.
 `task: rerun` reruns the most-recently spawned task. You can also rerun tasks from task modal.
 
+If you have indefinitely-running tasks that you need to restart on `task: rerun`, set `allow_concurrent_runs` to `true` and `use_new_terminal` to `false`.
```
Comment (SomeoneToIgnore) [inline docs/src/tasks.md:53]:
```suggestion
By default, rerunning tasks reuses the same terminal (due to the "use_new_terminal": false default) but wait for the previous task to finish before start (due to the "allow_concurrent_runs": false default).

Keep "use_new_terminal": false and set "allow_concurrent_runs": true to allow cancelling previous tasks on rerun.
```

**Comment by SomeoneToIgnore** [comment]
Gets better, thank you, but has nothing to do with infinite tasks.
At least now I have understood the way I want it to be phrased, so one last step and let's merge this.

**Comment by markuswustenberg** [comment]
@SomeoneToIgnore I fixed a typo and added some inline code formatting. What do you think?

**Review by SomeoneToIgnore — APPROVED** [review]
I've omitted the formatting because it was rendering odd for me in the suggestion preview but great I was wrong.

Thank you for bearing with me on this one!

**Pull request merged**